app/editor/src/features/admin/filters/utils/filterImportExport.ts

```diff
@@ -0,0 +1,208 @@
+  }
+  if ('inStory' in currentSettings) {
+    exportSettings = { ...exportSettings, inStory: currentSettings.inStory };
+  }
+  if ('inByline' in currentSettings) {
+    exportSettings = { ...exportSettings, inByline: currentSettings.inByline };
+  }
+  if ('edition' in currentSettings) {
+    exportSettings = { ...exportSettings, edition: currentSettings.edition };
+  }
+  if ('section' in currentSettings) {
+    exportSettings = { ...exportSettings, section: currentSettings.section };
+  }
+  if ('page' in currentSettings) {
+    exportSettings = { ...exportSettings, page: currentSettings.page };
+  }
+  if ('contentTypes' in currentSettings) {
+    exportSettings = { ...exportSettings, contentTypes: currentSettings.contentTypes };
+  }
+  if ('tags' in currentSettings) {
+    exportSettings = { ...exportSettings, tags: currentSettings.tags };
+  }
+  if ('sentiment' in currentSettings) {
+    exportSettings = { ...exportSettings, sentiment: currentSettings.sentiment };
+  }
+  if ('dateOffset' in currentSettings) {
+    exportSettings = { ...exportSettings, dateOffset: currentSettings.dateOffset };
+  } else {
+    if ('startDate' in currentSettings) {
+      exportSettings = { ...exportSettings, startDate: currentSettings.startDate };
+    }
+    if ('endDate' in currentSettings) {
+      exportSettings = { ...exportSettings, endDate: currentSettings.endDate };
+    }
+  }
+
+  // check the import JSON for currentSettingss which may need to be mapped
+  if ('actions' in currentSettings) {
+    const actionNames = actions
+      .filter((x) => currentSettings.actions.some((y: IActionModel) => y.id === x.id))
+      .map((x) => x.name);
+    exportSettings = { ...exportSettings, actionNames: actionNames };
+  }
+  if ('seriesIds' in currentSettings) {
+    const seriesNames = series
+      .filter((x) => currentSettings.seriesIds.some((y: number) => y === x.id))
+      .map((x) => x.name);
+    exportSettings = { ...exportSettings, seriesNames: seriesNames };
+  }
+  if ('sourceIds' in currentSettings) {
+    const sourceCodes = sources
+      .filter((x) => currentSettings.sourceIds.some((y: number) => y === x.id))
+      .map((x) => x.code);
+    exportSettings = { ...exportSettings, sourceCodes: sourceCodes };
+  }
+  if ('productIds' in currentSettings) {
+    const productNames = products
+      .filter((x) => currentSettings.productIds.some((y: number) => y === x.id))
+      .map((x) => x.name);
+    exportSettings = { ...exportSettings, productNames: productNames };
+  }
+  if ('contributorIds' in currentSettings) {
+    const contributorNames = contributors
+      .filter((x) => currentSettings.contributorIds.some((y: number) => y === x.id))
+      .map((x) => x.name);
+    exportSettings = { ...exportSettings, contributorNames: contributorNames };
+  }
+  exportFilter.settings = exportSettings;
+
+  return exportFilter;
+};
+
+export const parseExportedFilter = (
```

Comment:
Separate functions into their own files.  It's difficult to find things if they're all buried in one long file.